### PR TITLE
Add PCI Express 6.0 Header Support

### PR DESCRIPTION
**Pull request by sachinami**
# Description

PCI Express 6.0 Specification introduces new registers and modifies fields in existing ones.

This PR syncs MdePkg PciE headers with the spec update.

- [ ] Breaking change
- [ ] Impacts security
- [ ] Includes tests

## How This Was Tested

N/A

## Integration Instructions

N/A

Cc: Sergiy Yakovlev <sergiyy@ami.com>
Cc: Felix Polyudov <felixp@ami.com>
Cc: Dhanaraj V <vdhanaraj@ami.com>
Cc: Ray Ni <ray.ni@intel.com> @niruiyu 
Cc: Zhichao Gao <zhichao.gao@intel.com> @ZhichaoGao 
Cc: Liming Gao <gaoliming@byosoft.com.cn> @lgao4

### CODE REVIEW

File: MdePkg/Include/IndustryStandard/PciExpress60.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+Support for the PCI Express 6.0 standard.
+
+This header file may not define all structures.  Please extend as required.
+
+Copyright (c) 2024, American Megatrends International LLC. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _PCIEXPRESS60_H_
```
[the remark this answers is not in the document]
Comment (sachinami) [reply MdePkg/Include/IndustryStandard/PciExpress60.h]: Thank you @leiflindholm. It has been corrected and CI passes now.

File: MdePkg/Include/IndustryStandard/PciExpress21.h [hunk 2]
```diff
@@ -228,20 +236,21 @@ typedef union {
...
+    UINT16    ConfigurationRrsSoftwareVisibility : 1;
+    UINT16    NoNfmSubtree                       : 1;
+    UINT16    Reserved                           : 10;
   } Bits;
   UINT16    Uint16;
 } PCI_REG_PCIE_ROOT_CONTROL;
 
 typedef union {
   struct {
-    UINT16    CrsSoftwareVisibility : 1;
```
Comment (lgao4) [inline MdePkg/Include/IndustryStandard/PciExpress21.h:243]: Why change this field name? What impact caused by this change?

Comment (sachinami) [reply MdePkg/Include/IndustryStandard/PciExpress21.h]: @lgao4 The change was done to sync with the updated specification terminology. But prioritizing compatibility this has been reverted.

**Comment by sachinami** [comment]
@lgao4 If there are no concerns can this change be merged

File: MdePkg/Include/IndustryStandard/PciExpress21.h [hunk 3]
```diff
@@ -268,7 +277,7 @@ typedef union {
     UINT32    NoRoEnabledPrPrPassing                        : 1;
     UINT32    LtrMechanism                                  : 1;
     UINT32    TphCompleter                                  : 2;
-    UINT32    LnSystemCLS                                   : 2;
+    UINT32    Reserved                                      : 2;
```
Comment (lgao4) [inline MdePkg/Include/IndustryStandard/PciExpress21.h:280]: LnSystemCLS is not used any more?

File: MdePkg/Include/IndustryStandard/PciExpress21.h [hunk 4]
```diff
@@ -386,7 +416,7 @@ typedef struct {
   PCI_REG_PCIE_LINK_CAPABILITY2      LinkCapability2;
   PCI_REG_PCIE_LINK_CONTROL2         LinkControl2;
   PCI_REG_PCIE_LINK_STATUS2          LinkStatus2;
-  UINT32                             SlotCapability2;
+  PCI_REG_PCIE_SLOT_CAPABILITY2      SlotCapability2;
```
Comment (lgao4) [inline MdePkg/Include/IndustryStandard/PciExpress21.h:419]: This change will impact the source code. Have you meet with the real case with this field?

Comment (sachinami) [reply MdePkg/Include/IndustryStandard/PciExpress21.h:419]: Yes, this change is essential to support PciE Generation 6 specification. And it is inevitable.

File: MdePkg/Include/IndustryStandard/PciExpress21.h (hunk 3 above)
Comment (sachinami) [reply MdePkg/Include/IndustryStandard/PciExpress21.h:280]: Yes, Lightweight Notification has been deprecated and is not in use anymore.

File: MdePkg/Include/IndustryStandard/PciExpress21.h (hunk 4 above)
Comment (sachinami) [reply MdePkg/Include/IndustryStandard/PciExpress21.h:419]: The original field was intended as a placeholder till the capability was defined in the specification.

**Review by lgao4 — APPROVED** [review]

**Comment by alistair23** [comment]
Closing https://github.com/tianocore/edk2/pull/5753 as this was merged instead

**Comment by alistair23** [comment]
Wait, this PR misses DOE

**Pull request merged**